### PR TITLE
feat: fix adding obj options, adding possibility to write empty json map

### PR DESCRIPTION
**Pull request by asalzburger**
This PR fixes:
- missing `obj` options for `GeometryExample` leading to `boost::bad_any_cast`

It introduces:
- options for `json` geometry writing
- possibility to write an empty geometry map on demand to be filled in

The map would look like, e.g. for `--json-write-sf-layer` and the track ml detector:

```json
{
  "acts-geometry-hierarchy-map": {
    "format-version": 0,
    "value-identifier": "surface_types"
  },
  "entries": [
    {
      "volume": 5,
      "layer": 2,
      "value": "vol=5|lay=2"
    },
    {
      "volume": 7,
      "layer": 2,
      "value": "vol=7|lay=2"
    },
    {
      "volume": 7,
      "layer": 4,
      "value": "vol=7|lay=4"
    },
    {
      "volume": 7,
      "layer": 6,
      "value": "vol=7|lay=6"
    },
    {
      "volume": 7,
      "layer": 8,
      "value": "vol=7|lay=8"
    },
    {
      "volume": 7,
      "layer": 10,
      "value": "vol=7|lay=10"
    },
    {
      "volume": 7,
      "layer": 12,
      "value": "vol=7|lay=12"
    },
    {
      "volume": 7,
      "layer": 14,
      "value": "vol=7|lay=14"
    },
    {
      "volume": 8,
      "layer": 2,
      "value": "vol=8|lay=2"
    },
    {
      "volume": 8,
      "layer": 4,
      "value": "vol=8|lay=4"
    },
    {
      "volume": 8,
      "layer": 6,
      "value": "vol=8|lay=6"
    },
    {
      "volume": 8,
      "layer": 8,
      "value": "vol=8|lay=8"
    },
    {
      "volume": 9,
      "layer": 2,
      "value": "vol=9|lay=2"
    },
    {
      "volume": 9,
      "layer": 4,
      "value": "vol=9|lay=4"
    },
    {
      "volume": 9,
      "layer": 6,
      "value": "vol=9|lay=6"
    },
    {
      "volume": 9,
      "layer": 8,
      "value": "vol=9|lay=8"
    },
    {
      "volume": 9,
      "layer": 10,
      "value": "vol=9|lay=10"
    },
    {
      "volume": 9,
      "layer": 12,
      "value": "vol=9|lay=12"
    },
    {
      "volume": 9,
      "layer": 14,
      "value": "vol=9|lay=14"
    },
    {
      "volume": 10,
      "layer": 2,
      "value": "vol=10|lay=2"
    },
    {
      "volume": 12,
      "layer": 2,
      "value": "vol=12|lay=2"
    },
    {
      "volume": 12,
      "layer": 4,
      "value": "vol=12|lay=4"
    },
    {
      "volume": 12,
      "layer": 6,
      "value": "vol=12|lay=6"
    },
    {
      "volume": 12,
      "layer": 8,
      "value": "vol=12|lay=8"
    },
    {
      "volume": 12,
      "layer": 10,
      "value": "vol=12|lay=10"
    },
    {
      "volume": 12,
      "layer": 12,
      "value": "vol=12|lay=12"
    },
    {
      "volume": 13,
      "layer": 2,
      "value": "vol=13|lay=2"
    },
    {
      "volume": 13,
      "layer": 4,
      "value": "vol=13|lay=4"
    },
    {
      "volume": 13,
      "layer": 6,
      "value": "vol=13|lay=6"
    },
    {
      "volume": 13,
      "layer": 8,
      "value": "vol=13|lay=8"
    },
    {
      "volume": 14,
      "layer": 2,
      "value": "vol=14|lay=2"
    },
    {
      "volume": 14,
      "layer": 4,
      "value": "vol=14|lay=4"
    },
    {
      "volume": 14,
      "layer": 6,
      "value": "vol=14|lay=6"
    },
    {
      "volume": 14,
      "layer": 8,
      "value": "vol=14|lay=8"
    },
    {
      "volume": 14,
      "layer": 10,
      "value": "vol=14|lay=10"
    },
    {
      "volume": 14,
      "layer": 12,
      "value": "vol=14|lay=12"
    },
    {
      "volume": 16,
      "layer": 2,
      "value": "vol=16|lay=2"
    },
    {
      "volume": 16,
      "layer": 4,
      "value": "vol=16|lay=4"
    },
    {
      "volume": 16,
      "layer": 6,
      "value": "vol=16|lay=6"
    },
    {
      "volume": 16,
      "layer": 8,
      "value": "vol=16|lay=8"
    },
    {
      "volume": 16,
      "layer": 10,
      "value": "vol=16|lay=10"
    },
    {
      "volume": 16,
      "layer": 12,
      "value": "vol=16|lay=12"
    },
    {
      "volume": 17,
      "layer": 2,
      "value": "vol=17|lay=2"
    },
    {
      "volume": 17,
      "layer": 4,
      "value": "vol=17|lay=4"
    },
    {
      "volume": 18,
      "layer": 2,
      "value": "vol=18|lay=2"
    },
    {
      "volume": 18,
      "layer": 4,
      "value": "vol=18|lay=4"
    },
    {
      "volume": 18,
      "layer": 6,
      "value": "vol=18|lay=6"
    },
    {
      "volume": 18,
      "layer": 8,
      "value": "vol=18|lay=8"
    },
    {
      "volume": 18,
      "layer": 10,
      "value": "vol=18|lay=10"
    },
    {
      "volume": 18,
      "layer": 12,
      "value": "vol=18|lay=12"
    }
  ]
}
```

### CODE REVIEW

**Comment by asalzburger** [comment]
@niermann999 @noemina  - this will allow you to create an empty digitization map, which you only have to fill appropriately for the ITk.

File: Examples/Io/Csv/src/GeometryExampleBase.cpp [hunk 1]
```diff
@@ -0,0 +1,176 @@
+// This file is part of the Acts project.
```
Comment (Corentin-Allaire) [inline Examples/Io/Csv/src/GeometryExampleBase.cpp:1]: I think I am missing something about this file. Why is this a copy of GeometryExampleBase inside  Examples/Io/Csv ?

**Review by Corentin-Allaire — COMMENTED** [review]
Mostly uncontroversial but I don't understand the copy of GeometryExampleBase in Examples/Io/Csv.

File: Examples/Io/Csv/src/GeometryExampleBase.cpp (hunk 1 above)
Comment (asalzburger) [reply Examples/Io/Csv/src/GeometryExampleBase.cpp:1]: Ha, that's a mistake. Weird.

File: Examples/Run/Common/src/GeometryExampleBase.cpp [hunk 2]
```diff
@@ -38,10 +39,11 @@ int processGeometry(int argc, char* argv[],
...
   ActsExamples::Options::addCsvWriterOptions(desc);
-  ActsExamples::Options::addOutputOptions(desc,
-                                          ActsExamples::OutputFormat::Root |
-                                              ActsExamples::OutputFormat::Json |
-                                              ActsExamples::OutputFormat::Csv);
+  ActsExamples::Options::addJsonWriterOptions(desc);
+  ActsExamples::Options::addOutputOptions(
+      desc,
+      ActsExamples::OutputFormat::Root | ActsExamples::OutputFormat::Json |
+          ActsExamples::OutputFormat::Csv | ActsExamples::OutputFormat::Obj);
```
Comment (asalzburger) [inline Examples/Run/Common/src/GeometryExampleBase.cpp:46]: That's the bugfix, FYI.

Comment (Corentin-Allaire) [reply Examples/Run/Common/src/GeometryExampleBase.cpp:46]: Yes I saw that. Just to be sure, the ActsExamples::OutputFormat::Obj is not missing from any other Example file ?

Comment (asalzburger) [reply Examples/Run/Common/src/GeometryExampleBase.cpp:46]: I think we need to systematically go through the Examples, I will start doing this today.

Comment (Corentin-Allaire) [reply Examples/Run/Common/src/GeometryExampleBase.cpp:46]: Ok, I will approve this one at least !

**Review by Corentin-Allaire — APPROVED** [review]

File: Examples/Io/Csv/src/GeometryExampleBase.cpp (hunk 1 above)
Comment (asalzburger) [reply Examples/Io/Csv/src/GeometryExampleBase.cpp]: Deleted it.

**Pull request merged**